sympy/polys/polytools.py

```diff
@@ -6218,8 +6218,20 @@ def sqf_list(f, *gens, **args):
     (2, [(x + 1, 2), (x + 2, 3)])
 
     """
-    return _generic_factor_list(f, gens, args, method='sqf')
-
+    options.allowed_flags(args, ['frac', 'polys'])
```

Comment:
`_generic_factor_list` can handle also rational fractions if `frac=True`. It may then return factors with negative exponents. That does not apply to `sqf_list` so `'frac'` should be omitted. On the other hand, it might be useful to allow `'all'` that is accepted by the `.sqf_list` method. That should probably be removed from `args` before calling `allowed_flags` and saved (by something like `all = args.pop('all', False)`).

Reply:
okay, but what does adding `all = args.pop('all', False)` actually do?

Reply:
https://docs.python.org/3.7/library/stdtypes.html#dict.pop